src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java

```diff
@@ -484,34 +460,25 @@ private Lecture createLectureWithLectureUnits(HttpStatus expectedStatus) throws
             doc2.addPage(new PDPage());
             doc2.save(outputStream);
             MockMultipartFile file3 = new MockMultipartFile("file", "filename3.pdf", "application/json", outputStream.toByteArray());
-            unit3 = uploadAttachmentUnit(file3, expectedStatus);
+            lecture.getLectureUnits().add(uploadAttachmentUnit(lecture, file3, expectedStatus));
         }
 
-        lecture = lectureUtilService.addLectureUnitsToLecture(lecture, List.of(unit1, unit2, unit3));
+        // Collect unity freshly from the database to prevent issues when persisting the lecture again
```

Comment:
```suggestion
        // Collect units freshly from the database to prevent issues when persisting the lecture again
```
Or did you mean the party? :D